testing/README.md

```diff
@@ -0,0 +1,177 @@
+# Testing with Log4Cats
+
+## Why not just use [`NoOpLogger`](../noop/shared/src/main/scala/org/typelevel/log4cats/noop/NoOpLogger.scala)?
+
+While an error in production is unquestionably the most important time to have good logging, a more common time when
+having logs is really useful is figuring out why a test is failing. Unfortunately, logging and test reporting often
+aren't synchronized in a useful manner, and the output becomes difficult to decipher.
+
+A common response to this is simply to disable logging in the tests generally, and open it back up locally when
+debugging. `NoOpLogger` will certainly do that for you, however there is a better way - integrating log message into
+the output of _only_ failing tests.
+
+## How can [`org.typelevel.log4cats.testing`](shared/src/main/scala/org/typelevel/log4cats/testing/) help?
+
+How log4cats will be integrated into test output looks different depending on the test framework. An example of
+how to do this for `munit` could look like this:
+
+```scala
+package org.log4cats.example
+
+import cats.MonadThrow
+import cats.effect.unsafe.IORuntime
+import cats.effect.{IO, Resource}
+import cats.syntax.all._
+import munit.internal.console.Printers
+import munit.{CatsEffectSuite, Printer, TestOptions}
+import org.typelevel.log4cats.{LoggerFactory, SelfAwareStructuredLogger}
+import org.typelevel.log4cats.testing.StructuredTestingLogger
+import org.typelevel.log4cats.testing.StructuredTestingLogger.{DEBUG, ERROR, INFO, TRACE, WARN}
+
+import java.io.{PrintWriter, StringWriter}
+
+class ExampleTest extends CatsEffectSuite {
+  private val loggerFixture = ResourceFunFixture[LoggingHelper]((options: TestOptions) =>
+    Resource.make(IO.delay {
+      val logger = StructuredTestingLogger.impl[IO]()
+      LoggingHelper(logger, logger.addContext(Map("TestName" -> options.name)))
+    })(_ => IO.unit)
+  )
+
+  private val loggerPrinter = new Printer {
```

Comment:
There's rather a lot of boilerplate in this. It might be worth considering creating an integration for `munit`.

Reply:
We're on a milestone, for reasons I can't remember.  If that becomes a production dependency, we'd probably want a separate repo.